index.js

```diff
@@ -15,16 +15,23 @@ const getBloquequote = async (url) => {
 };
 
 module.exports = async ({ markdownAST }) => {
-  visit(markdownAST, 'text', async (node) => {
+  const nodes = [];
+  visit(markdownAST, 'text', (node) => {
     const { value } = node;
     const tweetLink = value.match(/https:\/\/twitter\.com\/([A-Za-z0-9-_]*\/status\/[A-Za-z0-9-_?=]*)/gi);
     if (tweetLink) {
-      console.log(`\n embeding tweet: ${tweetLink} \n`);
-      const embedData = await getBloquequote(tweetLink);
-      node.type = 'html';
-      node.value = embedData.html;
+      nodes.push([node, tweetLink]);
     }
-  });
+  })
+  for (let i = 0; i < nodes.length; i++) {
+    const nt = nodes[i];
+    const node = nt[0];
+    const tweetLink = nt[1];
+    console.log(`\n embeding tweet: ${tweetLink} \n`);
+    const embedData = await getBloquequote(tweetLink);
```

Comment:
Thanks @isaacs! I've built my own solution for this (needed more out-of scope features). In my solution I fire off all promises at the same time:

https://github.com/kentcdodds/kentcdodds.com/blob/037c9ac55ab59d1b7216a424b79024b857e626f4/plugins/remark-embedder/index.js#L8-L22

May be a good idea to use https://github.com/sindresorhus/p-throttle to limit it to 8 at once or something, but having these run at the same time will probably speed this up a lot.